module/metrics/unicast_manager.go

```diff
@@ -0,0 +1,137 @@
+package metrics
+
+import (
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+
+	"github.com/onflow/flow-go/module"
+)
+
+// UnicastManagerMetrics metrics collector for the unicast manager.
+type UnicastManagerMetrics struct {
+	// createStreamAttempts tracks the number of retry attempts to create a stream.
+	createStreamAttempts *prometheus.HistogramVec
+	// createStreamDuration tracks the overall time it takes to create a stream, this time includes
+	// time spent dialing the peer and time spent connecting to the peer and creating the stream.
+	createStreamDuration *prometheus.HistogramVec
+	// dialPeerAttempts tracks the number of retry attempts to dial a peer during stream creation.
+	dialPeerAttempts *prometheus.HistogramVec
+	// dialPeerDuration tracks the time it takes to dial a peer and establish a connection.
+	dialPeerDuration *prometheus.HistogramVec
+	// establishStreamOnConnAttempts tracks the number of retry attempts to create the stream after peer dialing completes and a connection is established.
+	establishStreamOnConnAttempts *prometheus.HistogramVec
+	// establishStreamOnConnDuration tracks the time it takes to create the stream after peer dialing completes and a connection is established.
+	establishStreamOnConnDuration *prometheus.HistogramVec
+
+	prefix string
+}
+
+var _ module.UnicastManagerMetrics = (*UnicastManagerMetrics)(nil)
+
+func NewUnicastManagerMetrics(prefix string) *UnicastManagerMetrics {
+	uc := &UnicastManagerMetrics{prefix: prefix}
+
+	uc.createStreamAttempts = promauto.NewHistogramVec(
```

Comment:
I think a counter vector seems more appropriate for this metric than histogram, sorry for the initial suggestion of histogram.